### PR TITLE
Course Sync Pages Interactor

### PR DESCRIPTION
**Pull request by szabinst**
This PR includes:
- container logic for downloading all kinds of content for a course
- pages download logic

---

refs: MBL-16721
affects: Student
release note: none
test plan: 
1. Have internet on 
2. Enable `offline_mode` feature flag
3. Open Course Sync menu 
4. Select Pages for download.
5. Turn off internet 
6. Open course, go to pages 
7. You should see a list of pages 

## Checklist

- [ ] Follow-up e2e test ticket created or not needed
- [x] A11y checked
- [x] Tested on phone
- [x] Tested on tablet
- [x] Tested in dark mode
- [x] Tested in light mode
- [ ] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Core/Core/CourseSync/CourseSyncDownloader/CourseSyncPagesInteractorPreview.swift:28:1: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line; currently 2 (vertical_whitespace)
    
    
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ecca8b42b35a1b0b646dc0121483c1013e8d2b50
</p>

**Review by vargaat — CHANGES_REQUESTED** [review]
- [x] `CourseSyncInteractor` has no unit tests.

File: Core/Core/CourseSync/CourseSyncSelector/ViewModel/CourseSyncSelectorViewModel.swift [hunk 1]
```diff
@@ -74,7 +83,8 @@ class CourseSyncSelectorViewModel: ObservableObject {
         interactor
             .observeIsEverythingSelected()
             .map { $0 ? NSLocalizedString("Deselect All", comment: "")
-                      : NSLocalizedString("Select All", comment: "") }
+                : NSLocalizedString("Select All", comment: "")
```
[the remark this answers is not in the document]
Comment (vargaat) [reply Core/Core/CourseSync/CourseSyncSelector/ViewModel/CourseSyncSelectorViewModel.swift:86]: As long as the two strings are starting at the same position I'm fine. Is this Xcode's default indentation?

[the remark this answers is not in the document]
Comment (szabinst) [reply Core/Core/CourseSync/CourseSyncSelector/ViewModel/CourseSyncSelectorViewModel.swift:86]: If the two strings are on a new line, then yes.

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Review by vargaat — APPROVED** [review]
QA+1

**Review by vargaat — DISMISSED** [review]
QA+1

**Review by idevainstr — APPROVED** [review]
QA +1

**Pull request merged**